lib/RSAKeyPair.js

```diff
@@ -38,50 +38,23 @@ class RSAKeyPair extends LDKeyPair {
    * @returns {Promise<RSAKeyPair>}
    */
   static async generate(options = {}) {
-    if(env.nodejs) {
-      const {generateKeyPair} = require('crypto');
-      // node >= 10.12.0
-      if(generateKeyPair) {
-        const {promisify} = require('util');
-        const pGenerateKeyPair = promisify(generateKeyPair);
-        const key = await pGenerateKeyPair('rsa', {
-          modulusLength: DEFAULT_RSA_KEY_BITS,
-          publicExponent: DEFAULT_RSA_EXPONENT,
-          publicKeyEncoding: {
-            type: 'spki',
-            format: 'pem'
-          },
-          privateKeyEncoding: {
-            type: 'pkcs8',
-            format: 'pem'
-          }
-        });
-        return new RSAKeyPair({
-          publicKeyPem: key.publicKey,
-          privateKeyPem: key.privateKey,
-          ...options
-        });
-      }
-    }
-
-    // browsers and node < 10.12.0
+    // forge will use a native implementation in nodejs >= 10.12.0
+    // and a purejs implementation in browser and nodejs < 10.12.0
     return new Promise((resolve, reject) => {
-      forge.pki.rsa.generateKeyPair(
-        {
-          bits: DEFAULT_RSA_KEY_BITS,
-          e: DEFAULT_RSA_EXPONENT,
-          workers: -1
-        },
-        (err, keyPair) => {
-          if(err) {
-            return reject(err);
-          }
-          resolve(new RSAKeyPair({
-            publicKeyPem: forge.pki.publicKeyToPem(keyPair.publicKey),
-            privateKeyPem: forge.pki.privateKeyToPem(keyPair.privateKey),
-            ...options
-          }));
-        });
+      forge.pki.rsa.generateKeyPair({
```

Comment:
Does the forge API not have a Promise interface natively, for that generateKeyPair call? Just curious, not a deal breaker.

Reply:
Forge hasn't been updated to use promises in general, no.